### ISSUE TITLE
Definition of args type of `where` in `SymbolTable`

### ISSUE THREAD

**Issue by whyfate** [FirelyTeam/firely-net-sdk#2553]
this is my test code.
```csharp
Hl7.FhirPath.Expressions.SymbolTableInit.AddStandardFP(FhirPathCompiler.DefaultSymbolTable);
FhirPathCompiler.DefaultSymbolTable.AddFhirExtensions();
var sr = new ServiceRequest
{
    Subject = new ResourceReference
    {
        Reference = "Patient/1",
    }
};

var context = FhirEvaluationContext.CreateDefault();
// Mock a ElementResolver only applicable to this unit test
context.ElementResolver = (s) =>
{
    return new Patient
    {
        Id = "1"
    }.ToTypedElement();
};
var values = sr.ToTypedElement().Select("ServiceRequest.subject.where(resolve() is Patient)", context);
Assert.AreEqual(1, values.Count());
var fhirValueProvider = values.First().Annotation<IFhirValueProvider>();
Assert.IsNotNull(fhirValueProvider);
Assert.IsNotNull(fhirValueProvider.FhirValue);
var r = fhirValueProvider.FhirValue as ResourceReference;
Assert.IsNotNull(r);
Assert.AreEqual("Patient/1", r.Reference);
```
When I don't change any code, this test doesn't pass，When I made the following changes, the test passed.
before:  
https://github.com/FirelyTeam/firely-net-sdk/blob/04bd24b50d843e2495a4b13b3fc27f2e610e4051/src/Hl7.Fhir.Base/FhirPath/Expressions/SymbolTableInit.cs#L202

after:
```csharp
t.Add(new CallSignature("where", typeof(IEnumerable<ITypedElement>), typeof(object), typeof(bool)), runWhere);
```
Is this modification correct?

**Comment by brianpos** [comment]
Which version of the SDK is that using?

**Comment by brianpos** [comment]
The existing code should be right, as that's how many of the search parameters work today and have done for some time.
[The where clause ](https://github.com/brianpos/fhirpath-lab-dotnet/blob/7768a68604fbe24311be0edabbcf6ac6434bc42b/fhirpath-lab-dotnet/FunctionFhirPathTest.cs#L235)
This example is using the firelySDK 5.2.0

```fhirpath
ServiceRequest.subject.trace('s').where(resolve().trace('p') is Patient)
```
:test_tube: [Test with FHIRPath-Lab](https://dev.fhirpath-lab.com/FhirPath?parameters=N4IgpgHgDgTmDO8CWB7AdiAXCAymGAbkgMZgBKYAjgK4IAuAdPNQEYBWYxjdMAhqQAoA5PCEBKBgHcAFvjAC48FABsC8iT37yhUcQAIk8PQAVedJGDR0xIADQhFKajFJYQp85boB6SLwC2UMpgduBoAOZIaCHYDAByAKIAKnoCAGZIcMoAnjb2dPj+USoo4dl4hPhu0nR0UPCY3t7wlMroadKZALQwACwMvABezmCSYCzIBfAM0T4dmaEEvDBIvCzB8FgA2gC69o7OpABSShjYwAA6DghOLmBJ2VBgV5hXFUSkFDT0V7ZXSAATF5XPyBYK-K7MdicOgvS7XNJyNCkYEgGp1BpNFptNDzGA9fpDEZjCZIKYzMBzTowbweCxWXwQAJBZ52K4AwxBXjZVEAYWkvGU-nw8FsJkp+D0RwCCCuAF85SA5UA)

**Comment by brianpos** [comment]
p.s. That specific test I get that issue too. weird.
I don't understand what's different.

**Comment by whyfate** [comment]
I debugged the code and found that it was a problem with this method.
https://github.com/FirelyTeam/firely-net-sdk/blob/04bd24b50d843e2495a4b13b3fc27f2e610e4051/src/Hl7.Fhir.Base/FhirPath/Expressions/SymbolTable.cs#L102-L111

Did the relative URI and absolute URI of the resource use different processing methods?

**Comment by whyfate** [comment]
[Test with FHIRPath-Lab by relative URI](https://dev.fhirpath-lab.com/FhirPath?expression=ServiceRequest.subject.trace('s').where(resolve().trace('p')%20is%20Patient)&engine=.NET%20(firely)&resourceJson=N4KABGBEBOCmDOB7ArtAxrAKgTwA60gC4oBlWaANwEsMAlWAR2QQBdIAacKKgEyKlgAPAIYBbXABsCnCJHjIARgCtYaNsVARZcAGblYAOwz9IABWEsqhlgHohYydK6yeVeJOHYTAYQAWwiVFyeHYwU1gWcjAAKTEESC4AXxBEoA&terminologyserver=https%3A%2F%2Fsqlonfhir-r4.azurewebsites.net%2Ffhir)

**Comment by brianpos** [comment]
Yeah, it's running on the server, (kinda can't run the dotnet in the browser) so if it has the fullURL it's able to know where to look.

**Comment by whyfate** [comment]
```csharp
SymbolTableInit.AddStandardFP(FhirPathCompiler.DefaultSymbolTable);
FhirPathCompiler.DefaultSymbolTable.AddFhirExtensions();
var sr = new ServiceRequest
{
    Subject = new ResourceReference
    {
        Reference = "https://spark.incendi.no/fhir/Patient/1",
    }
};
var values = sr.ToTypedElement().Select("ServiceRequest.subject.where(resolve() is Patient)");
```
It's my local test code,SDK use `Hl7.Fhir.R4:5.3.0`,I use the fullURL but not return data,why？

**Comment by brianpos** [comment]
Can you roll back to the 5.2.0 release and see if this is a new issue (I haven't updated to that version yet)
The other difference I note in your code is taht you don't provide the FhirContext that takes your resource as a parameter.
When I did that, at least the resolve function was being called.

**Comment by whyfate** [comment]
Yeah, I still have this issue with version 5.2.0 of SDK.
![image](https://github.com/FirelyTeam/firely-net-sdk/assets/13515020/f8e14ff8-76d9-4253-aac4-377c5a4785c4)

**Comment by whyfate** [comment]
It's my fault.
I fancy oneself clever added this code.
```csharp
SymbolTableInit.AddStandardFP(FhirPathCompiler.DefaultSymbolTable);
```
Remove it and will be correct.
Thank you very much for your answer @brianpos .
